internal/services/storage/storage_account_network_rules_resource.go

```diff
@@ -184,12 +177,74 @@ func resourceStorageAccountNetworkRulesCreateUpdate(d *pluginsdk.ResourceData, m
 			NetworkAcls: acls,
 		},
 	}
-	if _, err := client.Update(ctx, *id, payload); err != nil {
+	if _, err = client.Update(ctx, *id, payload); err != nil {
 		return fmt.Errorf("updating Network Rules for %s: %+v", *id, err)
```

Comment:
Although it's an update should the error message say creating?
```suggestion
		return fmt.Errorf("creating Network Rules for %s: %+v", *id, err)
```